### PR TITLE
First draft of language detection (untested)

### PR DESCRIPTION
**Pull request by aidanjungo**
tournesol-app/tournesol-meta#58

I only tested the function 'languages_detection' for the rest, I don't have the possibility to test it locally.

### CODE REVIEW

File: tournesol/utils/video_language.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+def languages_detection(title, description):
+    ''' This algo make 18 errors / 1238 known video (1.45 % of error) '''
+
+    # Remove lines which contains links, mail or year
+    description = description.split('\n')
+    description = [line for line in description if not 'www' in line]
+    description = [line for line in description if not 'http' in line]
+    description = [line for line in description if not '@' in line]
+    description = [line for line in description if not re.search(r'^(17|20)\d{2}$', line)]
```
Comment (lfaucon) [inline tournesol/utils/video_language.py:15]: Simplify in one line: 

```Python
description = [line for line in description.split('\n') if not re.search(r'^(17|18|19|20)\d{2}|www|http|@$', line)]
```

And why not filtering words instead to not remove entire lines

**Review by lfaucon — CHANGES_REQUESTED** [review]
Amazing. It worked out of the box except for few errors.

I get a lot of Wraning "This video should be checked manually" out of few tests

The following is incorrectly found to be in English instead of German:
https://www.youtube.com/watch?v=TjqhsbGUoJw 

The following has None for the language instead of 'fr'
https://www.youtube.com/watch?v=HTmtkhh0AnU

The following lead to an error `"('fr',)" is not a valid choice.` probably language should be `'fr'` not `'('fr',)`.
https://www.youtube.com/watch?v=ZVYaGfs80b0
https://www.youtube.com/watch?v=VvW_g4gdgm4
This probably happens because of incorrectly reusing known languages. I can fix it.

File: tournesol/utils/video_language.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    except:
+        lang = None
+   
+    return lang
+
+
+def compute_video_language(channel, title, description):
+
+    # Get language(s) from other video of the same channel
+    lang_list = Video.objects.filter(uploader=channel).values_list("language")
```
Comment (lfaucon) [inline tournesol/utils/video_language.py:28]: `channel` is called `uploader` in the codebase

File: tournesol/utils/video_language.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+        main_channel_lang, main_channel_lang_cnt = Counter(lang_list).most_common()[0]
+        
+        # TODO: These value could be modified...
+        if len(lang_list) > 4 and main_channel_lang_cnt/len(lang_list) > 0.9 : 
+            return main_channel_lang
+
+    # TODO: Transform that in a log or a Warning...
+    print('The following video should be checked manually!!!') 
+    print(title)
```
Comment (lfaucon) [inline tournesol/utils/video_language.py:40]: We should not have this warning for every video that uses langdetect

File: tournesol/utils/video_language.py (hunk 2 above)
Comment (lfaucon) [reply tournesol/utils/video_language.py:28]: I see you took it from somewhere in the code that had this called "channel". Let's rename it there too :)

**Review by lfaucon — CHANGES_REQUESTED** [review]

**Comment by lfaucon** [comment]
I pushed a commit. I think one of the main improvement point would be to not remove entire lines if they contain one incorrect word, instead you may split on blank spaces and filter word per word.

File: requirements.txt [hunk 4]
```diff
@@ -16,6 +16,7 @@ psycopg2-binary==2.8.6
 pytest==6.2.4
 PyYAML==5.4.1
 tqdm==4.62.1
+langdetect==1.0.8
```
Comment (lfaucon) [inline requirements.txt:19]: Let's add a paragraph in python-dependency.md about this package. An a link to its doc :)

**Review by lfaucon — CHANGES_REQUESTED** [review]

File: tournesol/utils/video_language.py (hunk 2 above)
Comment (aidanjungo) [reply tournesol/utils/video_language.py]: Done!

File: tournesol/utils/video_language.py (hunk 3 above)
Comment (aidanjungo) [reply tournesol/utils/video_language.py]: OK, I removed it.

File: requirements.txt (hunk 4 above)
Comment (aidanjungo) [reply requirements.txt:19]: Done!

File: tournesol/utils/video_language.py (hunk 1 above)
Comment (aidanjungo) [reply tournesol/utils/video_language.py]: Because it is better to remove publications (often in English) which are in description.

**Pull request merged**